File: assets/JavaScript/UI/EventListeners/sidebar_toggle.mjs


function toggle(n_clicks, current_data_val){
    if (current_data_val === 1){return 0;}
    else {return 1;}
}

window.dash_clientside = Object.assign(
    {},
    window.dash_clientside, {
        'sidebar_toggle': {
            'toggle': toggle
        }
    }
);

File: Components/UIComponents/EventListeners/ClientsideListeners/sidebar_toggle.py
# Set shebang if needed
# -*- coding: utf-8 -*-
"""
Created on Wed Feb  4 16:15:09 2026

@author: mano
"""

from dash import clientside_callback, ClientsideFunction, Input, State, Output

def add_sidebar_toggle_event_listener():

    clientside_callback(
        ClientsideFunction(
            namespace='sidebar_toggle',
            function_name='toggle'
        ),
        Output('app-body', 'data-checked'),
        Input('hamburger-btn-label', 'n_clicks'),
        State('app-body', 'data-checked'),
        prevent_initial_call=True
    )
    return None